gmso/utils/units.py

```diff
@@ -79,12 +76,426 @@ def default_reg():
+            * energySym ** (-1 * energy_exp)
+        )
+
+    @staticmethod
+    def _dimensions_to_charge(dims):
+        """Take a set of dimensions and substitute in Symbol("charge") where possible."""
+        symsStr = str(dims.free_symbols)
+        charge_inBool = np.all(
+            [dimStr in symsStr for dimStr in ["current_mks"]]
+        )
+        if not charge_inBool:
+            return dims
+        chargeSym = Symbol(
+            "(charge)"
+        )  # create dummy symbol to replace in equation
+        dim_info = dims.as_terms()
+        current_idx = np.where(
+            list(map(lambda x: x.name == "(current_mks)", dim_info[1]))
+        )[0][0]
+        charge_exp = dim_info[0][0][1][1][
+            current_idx
+        ]  # charge has (current_mks) in it, so this is the hint of how many
+        return (
+            dims
+            * u.dimensions.charge ** (-1 * charge_exp)
+            * chargeSym**charge_exp
+        )
+
+    @staticmethod
+    def _dimensions_from_thermal_to_energy(dims):
+        """Take a set of dimensions and substitute in Symbol("energy") to replace temperature."""
+        symsStr = str(dims.free_symbols)
+        temp_inBool = np.all([dimStr in symsStr for dimStr in ["temperature"]])
+        if not temp_inBool:
+            return dims
+        energySym = Symbol(
+            "(energy)"
+        )  # create dummy symbol to replace in equation
+        dim_info = dims.as_terms()
+        temp_idx = np.where(
+            list(map(lambda x: x.name == "(temperature)", dim_info[1]))
+        )[0][0]
+        temp_exp = dim_info[0][0][1][1][
+            temp_idx
+        ]  # energy has 1/time**2 in it, so this is the hint of how many
+        return (
+            dims
+            / u.dimensions.temperature**temp_exp
+            * energySym ** (temp_exp)
+        )
+
+    @classmethod
+    def _get_output_dimensions(cls, dims, thermal_equivalence=False):
+        if str(dims) == "1":  # use string as all dims can be converted
+            return u.dimensionless
+        dims = cls._dimensions_to_energy(dims)
+        dims = cls._dimensions_to_charge(dims)
+        if thermal_equivalence:
+            dims = cls._dimensions_from_thermal_to_energy(dims)
+        return dims
+
+
+class ljUnitSystem:
+    """Use this so the empty unitsystem has getitem magic method."""
+
+    def __init__(self, reg: u.UnitRegistry):
+        self.registry = reg
+        self.name = "lj"
+
+    def __getitem__(self, item):
+        """Return dimensionless units unless angle."""
+        if item == "angle":
+            return u.Unit("degree")
+        return u.Unit("dimensionless")
+
+
+def get_parameter_dimension(parameter, dimension):
+    """Return a unit from the parameter in a given dimension."""
+    param_terms = parameter.units.expr.as_terms()
+    uStr = ""
+    for symbol, exp in zip(param_terms[-1], param_terms[0][0][1][1]):
+        outputDim = LAMMPS_UnitSystems._get_output_dimensions(
```

Comment:
Would it make more sense to move the `_get_output_dimensions` (and its internal workers methods, i.e., `_dimension_to_energy` and `_dimension_to_charge`) to be outside of the `LAMMPS_UnitSystems`, since they are class/static methods anyway, and is now being used on a function outside of the `LAMMPS_UnitSystems` (that's not LAMMPS specific)

Reply:
I think it will be yes. The way I viewed it, we will refactor this in one more step to make a generic GMSO_UnitSystems class, with some of the more general units methods. Then LAMMPS_UnitSystems can only contain the methods that are specific to it, i.e. the `usystem_from_str` function, and maybe also keep the `convert_parameter` function engine specific as well. I think during that step, we can make these _get_output_dimension just a set of functions in the module. But I'd like to do that whole refactor in a future PR.